app/src/main/java/org/wikipedia/page/PageActivity.kt

```diff
@@ -391,9 +393,9 @@ class PageActivity : BaseActivity(), PageFragment.Callback, LinkPreviewDialog.Ca
     }
 
     override fun onPageCloseActionMode() {
-        val actionModesToFinish = HashSet(currentActionModes)
-        for (mode in actionModesToFinish) {
-            mode.finish()
+        val actionModesToFinish = ArraySet(currentActionModes)
+        for (i in actionModesToFinish.indices) {
```

Comment:
Why not keep the previous (cleaner) syntax of `for (mode in actionModesToFinish)`?

Reply:
The documentation [recommends](https://developer.android.com/reference/androidx/collection/ArraySet#iterator()) against using the iterator.